grails-datastore-gorm-hibernate-core/src/main/groovy/org/grails/orm/hibernate/cfg/AbstractGrailsDomainBinder.java

```diff
@@ -1950,6 +1943,13 @@ else if (collectionType != null) {
                     value = collection;
                 }
             }
+            else if (userType != null) {
+                if (LOG.isDebugEnabled()) {
+                    LOG.debug("[GrailsDomainBinder] Binding property [" + currentGrailsProp.getName() + "] as SimpleValue");
+                }
+                value = new SimpleValue(mappings, table);
+                bindSimpleValue(currentGrailsProp, null, (SimpleValue) value, EMPTY_PATH, mappings, sessionFactoryBeanName);
+            }
             else if (currentGrailsProp.getType().isEnum()) {
```

Comment:
If it's not immediately obvious, all I have done is switch the "if" and "else if" here so that the collectionType check comes first and the userType check comes second. Necessary because both userType and collectionType are not null when using a UserCollectionType as the "type" in the mapping.